certbot/main.py

```diff
@@ -669,6 +669,16 @@ def main(cli_args=sys.argv[1:]):
 
     sys.excepthook = functools.partial(_handle_exception, config=config)
 
+    # Avoid conflicting args
+    conficting_args = ["quiet", "noninteractive_mode", "text_mode"]
+    if config.dialog_mode:
+        for arg in conficting_args:
+            if getattr(config, arg):
+                raise errors.Error(
+                    ("Conflicting values for displayer."
+                    " {0} conflicts with dialog_mode").format(arg)
+                )
```

Comment:
Sorry I didn't catch this last night, but this code is really just part of command line parsing. We should probably move it to `parse_args` (or a function/method called by `parse_args`) in `cli.py` instead of `main.py`. This allows you to move your test back in to `cli.py`, mitigating your code duplication problem.